spec/factories/asset_factory.rb

```diff
@@ -87,6 +91,14 @@
         faked_width { nil }
       end
 
+      trait :mp3_with_correct_sha512 do
+        faked_file { File.open((Rails.root + "spec/test_support/audio/ice_cubes.mp3")) }
+        faked_sha512 { Digest::SHA512.file(Rails.root + "spec/test_support/audio/ice_cubes.mp3").to_s }
+        faked_content_type { "audio/mpeg" }
+        faked_height { nil }
+        faked_width { nil }
+      end
+
```

Comment:
I think _maybe_ you could do with this our existing traits/factories, but it's confusing adding a new one is fine! But I think this should be a sub-factory not a trait. You can't really combine it with other traits in ways that makes sense, just make it a sub-factory.

Reply:
Sub-factories! What a concept!

Reply:
Fixed